Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Shared
+import Storage
+import UIKit
+import WebKit
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+// MARK: - OpenSearch
+
+public struct OpenSearchReference: Codable {
+    let reference: String
+    let title: String?
+
+    private enum CodingKeys: String, CodingKey {
+        case reference = "href"
+        case title = "title"
+    }
+}
+
+// MARK: - OpenSearch Browser Extension
+
+extension BrowserViewController {
+    
+    /// Adding Toolbar button over the keyboard for adding Open Search Engine
+    /// - Parameter webView: webview triggered open seach engine
+    func evaluateWebsiteSupportOpenSearchEngine(_ webView: WKWebView) {
+        if let openSearchMetaData = tabManager.selectedTab?.pageMetadata?.search {
```

Comment:
Since you have the `webView`,  perhaps we can obtain the tab using it instead of relying on `selectedTab`:

```suggestion
        if let tab = tabManager[webView], let openSearchMetaData = tab.pageMetadata?.search {
```

Reply:
Fix at https://github.com/brave/brave-ios/pull/3231/commits/caf9330e200b9172208e804ce5f38fd5720089a8